sailthru/Sailthru_Client.php

```diff
@@ -1420,37 +1428,47 @@ protected function httpRequestCurl($url, array $data, $method = 'POST', $options
             }
         }
         curl_setopt($ch, CURLOPT_URL, $url);
-        curl_setopt($ch, CURLOPT_HEADER, false);
+        curl_setopt($ch, CURLOPT_HEADER, true);
         curl_setopt($ch, CURLOPT_ENCODING, 'gzip');
         curl_setopt($ch, CURLOPT_TIMEOUT_MS, $options['timeout']);
         curl_setopt($ch, CURLOPT_CONNECTTIMEOUT_MS, $options['connect_timeout']);
         curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
 
         curl_setopt($ch, CURLOPT_HTTPHEADER, $this->httpHeaders);
-        $data = curl_exec($ch);
+        $response = curl_exec($ch);
         $this->lastResponseInfo = curl_getinfo($ch);
+
         curl_close($ch);
-        if (!$data) {
+
+        // parse headers and body
+        $parts = explode("\r\n\r\nHTTP/", $response);
```

Comment:
i assume you got this from [this stackoverflow question](http://stackoverflow.com/a/17971689/396193)? the CURLOPT_HEADERFUNCTION looks kind of cool too, but works for me